### PR TITLE
Add SEG to dedicated VS. Remove FQDN in case of wrong certificate

### PR DESCRIPTION
**Pull request by akshayhavile**
Add SEG to dedicated VS. Remove FQDN in case of wrong certificate.

### CODE REVIEW

**Comment by akshayhavile** [comment]
build ako

File: internal/nodes/avi_model_evh_nodes.go [hunk 1]
```diff
@@ -2014,11 +2020,13 @@ func buildWithInfraSettingForEvh(key string, vs *AviEvhVsNode, vsvip *AviVSVIPNo
...
 	}
 }
-func DeleteDedicatedEvhVSNode(vsNode *AviEvhVsNode, key string) {
+func DeleteDedicatedEvhVSNode(vsNode *AviEvhVsNode, key string, hostsToRemove []string) {
 	vsNode.PoolGroupRefs = []*AviPoolGroupNode{}
 	vsNode.PoolRefs = []*AviPoolNode{}
 	vsNode.HttpPolicyRefs = []*AviHttpPolicySetNode{}
 	vsNode.DeletSSLRefInDedicatedNode(key)
+	RemoveFqdnFromEVHVIP(vsNode, hostsToRemove, key)
+	//RemoveFqdnFromVIP(vsNode, key, hostsToRemove)
```
Comment (monotosh-avi) [inline internal/nodes/avi_model_evh_nodes.go:2029]: commented out code

**Review by monotosh-avi — COMMENTED** [review]
One minor comment

File: internal/nodes/avi_model_evh_nodes.go (hunk 1 above)
Comment (akshayhavile) [reply internal/nodes/avi_model_evh_nodes.go]: Removed.

**Review by monotosh-avi — DISMISSED** [review]

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Review by sudswasavi — DISMISSED** [review]

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

File: internal/nodes/avi_model_evh_nodes.go [hunk 2]
```diff
@@ -807,6 +807,8 @@ func (o *AviObjectGraph) ConstructAviL7SharedVsNodeForEvh(vsName, key string, ro
 
 	if infraSetting := routeIgrObj.GetAviInfraSetting(); infraSetting != nil {
 		buildWithInfraSettingForEvh(key, avi_vs_meta, vsVipNode, infraSetting)
+	} else if dedicated {
+		avi_vs_meta.ServiceEngineGroup = lib.GetSEGName()
```
Comment (chauhanshubham) [inline internal/nodes/avi_model_evh_nodes.go:811]: isn't the `ServiceEngineGroup` being set in line L760?

File: internal/nodes/avi_model_l7_translator.go [hunk 3]
```diff
@@ -126,6 +126,8 @@ func (o *AviObjectGraph) ConstructAviL7VsNode(vsName string, key string, routeIg
 
 	if infraSetting := routeIgrObj.GetAviInfraSetting(); infraSetting != nil {
 		buildWithInfraSetting(key, avi_vs_meta, vsVipNode, infraSetting)
+	} else if dedicatedVs {
+		avi_vs_meta.ServiceEngineGroup = lib.GetSEGName()
```
Comment (chauhanshubham) [inline internal/nodes/avi_model_l7_translator.go:130]: Same here. L71 in this function.

File: internal/nodes/avi_model_evh_nodes.go (hunk 2 above)
Comment (akshayhavile) [reply internal/nodes/avi_model_evh_nodes.go]: Nice catch. Fixed at both places.

**Review by chauhanshubham — APPROVED** [review]

**Review by monotosh-avi — APPROVED** [review]

**Pull request merged**